app/Repositories/Organization/OrganizationRepository.php

```diff
@@ -0,0 +1,413 @@
+        $organizationAdminUserIds = $organization->users()->wherePivot('organization_role_type_id', 1)->get()->modelKeys();
+        $parentOrganizationUserIds = $this->getParentOrganizationUserIds([], $organization, $organizationAdminUserIds);
+
+        if ($data['page'] === 'create') {
+            $userInIds = array_merge($organizationUserIds, $parentOrganizationUserIds);
+        } else if ($data['page'] === 'update') {
+            $userInIds = array_diff($parentOrganizationUserIds, $organizationUserIds);
+        }
+
+        $this->query = $this->userRepository->model->when($data['query'] ?? null, function ($query) use ($data) {
+            $query->search(['email'], $data['query']);
+            return $query;
+        });
+
+        return $this->query->whereIn('id', $userInIds)->orderBy('first_name', 'asc')->paginate();
+    }
+
+    /**
+     * Get the parent organizations user ids
+     *
+     * @param $userIds
+     * @param $organization
+     * @param $organizationAdminUserIds
+     * @return array
+     */
+    public function getParentOrganizationUserIds($userIds, $organization, $organizationAdminUserIds)
+    {
+        if ($parentOrganization = $organization->parent) {
+            $ids = $parentOrganization->users()->wherePivot('organization_role_type_id', '<>', 1)->get()->modelKeys();
+            $adminIds = $parentOrganization->users()->wherePivot('organization_role_type_id', 1)->get()->modelKeys();
+            $ids = array_diff($ids, $organizationAdminUserIds);
+            $organizationAdminUserIds = array_merge($organizationAdminUserIds, $adminIds);
+            $userIds = array_merge($userIds, $ids);
+
+            $userIds = $this->getParentOrganizationUserIds($userIds, $parentOrganization, $organizationAdminUserIds);
+        }
+
+        return $userIds;
+    }
+
+    /**
+     * Add user for the specified role in particular suborganization
+     *
+     * @param Organization $organization
+     * @param array $data
+     * @return Model
+     */
+    public function addUser($organization, $data)
+    {
+        try {
+            $organization->users()->attach($data['user_id'], ['organization_role_type_id' => $data['role_id']]);
+
+            return true;
+        } catch (\Exception $e) {
+            Log::error($e->getMessage());
+        }
+
+        return false;
+    }
+
+    /**
+     * Add role for particular organization
+     *
+     * @param Organization $organization
+     * @param array $data
+     * @return Model
+     */
+    public function addRole($organization, $data)
+    {
+        try {
+            DB::beginTransaction();
+
+            $role = $organization->roles()->create([
+                'name' => $data['name'],
+                'display_name' => $data['display_name'],
+            ]);
+
+            $role->permissions()->attach($data['permissions']);
+
+            DB::commit();
+
+            return true;
+        } catch (\Exception $e) {
```

Comment:
Perhaps, there are two many blank lines.
See code in other places as well.

Reply:
"DB::beginTransaction();" and "DB::commit();" are a pair, I like to keep them on new line for better readability.